airbyte-integrations/connectors/destination-snowflake/src/test-integration/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/AbstractSnowflakeSqlGeneratorIntegrationTest.kt

```diff
@@ -1857,20 +1856,5 @@ abstract class AbstractSnowflakeSqlGeneratorIntegrationTest :
         private var dataSource: DataSource =
             SnowflakeDatabaseUtils.createDataSource(config, OssCloudEnvVarConsts.AIRBYTE_OSS)
         private var database: JdbcDatabase = SnowflakeDatabaseUtils.getDatabase(dataSource)
-
```

Comment:
This is interesting. Because the tests are now running in the same JVM and faster, we're now running 2 subclasses of `AbstractSnowflakeSqlGeneratorIntegrationTest`. Sometimes, the afterAll of one of those subclasses gets called while the other one is still running. And as a result, the 1st closes the dataSource (and all the connections and statements coming from it), causing the 2nd one to fail.
The original datasource (created in the `var` declaration (which should now be a val), wasn't getting `close`d, so removing the  `@AfterAll` doesn't change anything in that regard. And the `@BeforeAll` becomes unncecessary...

Reply:
The behavior was verified in the logs and was causing regular CI failures.